include/bluetooth/bluetooth.h

```diff
@@ -497,6 +523,18 @@ int bt_le_scan_start(const struct bt_le_scan_param *param, bt_le_scan_cb_t cb);
  */
 int bt_le_scan_stop(void);
 
+/** @brief Register a scanner packet listener.
+ *
+ *  Adds the listener to the list of scanner packet callbacks. The listeners
+ *  are called on every received scanner packet, regardless of what initiated
+ *  the scan.
+ *
+ *  @param listener Listener to register. Must point to static memory.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_scan_cb_register(struct bt_le_scan_listener *listener);
```

Comment:
Could we make the naming consistent here: `bt_le_<object>_register()` should take an parameter called `bt_le_<object>`. Now you have `scan_cb` for the first and `scan_listener` for the second.

Reply:
+1

Also after removing the k_mutex, we can also call this before `bt_init` so the check `if (!atomic_test_bit(bt_dev.flags, BT_DEV_READY))` could be removed as well. Which would be consistent with the rest of `bt_<thing>_register` functions.